File: Java/Labs/Module 1/Lab 1/Q2 - Contiguous Array/Main.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Main {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        String str=input.nextLine();
        ArrayList<Integer> nums=new ArrayList<Integer>();
        for(int i=0;i<str.length();i++){
            if(str.charAt(i)=='0'){
                nums.add(0);
            }
            else if(str.charAt(i)=='1'){
                nums.add(1);
            }
        }
        int ret = doSomething(nums);        
        System.out.println(ret);
    }
    
    public static int doSomething(ArrayList<Integer> nums) {
        int count=0;
        for(int i=0;i<nums.size();i++){
            int zero=0,one=0;
            for(int j=i;j<nums.size();j++){
                if(nums.get(j)==0){
                    zero++;
                }else{
                    one++;
                }
                if(zero==one){
                    count=Math.max(count,j-i+1);
                }
            }
        }
        return count;
    }
}
                                                                                                                            

File: Python/Labs/Module 1/Lab 1/Q2 - Contiguous Array/main.py
import sys
def doSomething(inval):
    count=0
    outval=0
    dict_seen={0:-1}
    for i in range(len(inval)):
        n=inval[i]
        if n==0:
            count-=1
        if n==1:
            count+=1
        if count in dict_seen:
            outval=max(outval,i-dict_seen[count])
        else:
            dict_seen[count]=i
    return outval
inputVal = input()    
newlist=inputVal.split(" ")
outputVal = doSomething(newlist)
print (outputVal)
                                                                                                                            